### PR TITLE
signer: Capture signer rejections and report them

### PR DESCRIPTION
**Pull request by cdecker**
We want to capture policy signer rejections too, as they can help us identify problems for false negatives, and can be used to fine-tune policies, or act as a regression test suite should the rejection be a true positive.

Sadly we were not actually sending a rejection, due to a misplaces `?` that'd bubble up the error past the reporting spot. This PR addresses that by handling the response in a more fine-grained manner.

### CODE REVIEW

File: libs/gl-client/src/signer/mod.rs [hunk 1]
```diff
@@ -539,15 +539,24 @@ impl Signer {
...
+                msg: format!("{:?}", e),
+                request: Some(req.clone()),
+                git_version: GITHASH.to_string(),
+            })
+            .await;
+            return Err(Error::Other(anyhow!("processing request: {e:?}")));
         }
-        .map_err(|e| Error::Other(anyhow!("processing request: {e:?}")))?;
 
+        let response = response.unwrap();
```
Comment (philipr-za) [inline libs/gl-client/src/signer/mod.rs:554]: You can avoid the clone by dropping the reporter call into the existing map_err closure:
```rust
.map_err(|e| 
    {
        report::Reporter::report(crate::pb::scheduler::SignerRejection {
                msg: format!("{:?}", e),
                request: Some(req.clone()),
                git_version: GITHASH.to_string(),
            })
            .await;
          Error::Other(anyhow!("processing request: {e:?}"))
    }
)?;

```

Comment (cdecker) [reply libs/gl-client/src/signer/mod.rs:554]: Clever ^^

Comment (cdecker) [reply libs/gl-client/src/signer/mod.rs:554]: Yeah, sadly we can't `.await` inside of a `map_err`, but would have been nice ^^

Comment (philipr-za) [reply libs/gl-client/src/signer/mod.rs:554]: Ahh, my async is rusty

**Review by philipr-za — APPROVED** [review]
Not massively well versed in this codebase yet but LGTM

**Review by Randy808 — APPROVED** [review]

**Pull request merged**